source/assets/js/playground.ts

```diff
@@ -68,11 +77,64 @@ function setupPlayground() {
     parent: document.querySelector('.sl-code-is-compiled') || undefined,
   });
 
+  /**
+   * Returns a playground state selection for the current single non-empty
+   * selection, or `null` otherwise.
+   */
+  function editorSelectionToStateSelection():
+    | PlaygroundState['selection']
+    | null {
+    const sel = editor.state.selection;
+    if (sel.ranges.length !== 1) {
+      return null;
+    }
+
+    const range = sel.ranges[0];
+    if (range.empty) {
+      return null;
+    }
+
+    const fromLine = editor.state.doc.lineAt(range.from);
+    const toLine = editor.state.doc.lineAt(range.to);
+
+    return [
+      fromLine.number,
+      range.from - fromLine.from + 1,
+      toLine.number,
+      range.to - toLine.from + 1,
+    ];
+  }
+
+  function restoreSelection() {
+    if (playgroundState.selection === null) {
+      return;
```

Comment:
Should this clear the editor's selection if it has one?

Reply:
hmmm 🤔 The editor's always at L1C1 when it loads, but yes... I think clearing the selection here makes that function consistent the other `update*` functions. I'll also rename it to `updateSelection`. Thanks! 😊